### DISCUSSION TITLE
Can I use ORM mode with SQLAlchemy and JSON fields for Postgres or Mysql?

### DISCUSSION THREAD

**Discussion by acidjunk** [fastapi/fastapi#7350] [Questions] [answered]
### First check

* [x ] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Relevant parts of DB Model (using declarative base) 

```Python
from sqlalchemy.dialects.mysql import JSON
from sqlalchemy import Column, Text 

class Certificate(Base):
    __tablename__ = "certificates"

    commonname = Column(JSON, default=[], nullable=False)
    certificate = Column(Text, nullable=True)
    cachain = Column(Text, nullable=True)
```

Schema
```Python
from typing import List, Optional

from pydantic import BaseModel, Json


# Shared properties
class CertificateBase(BaseModel):
    commonname: Optional[Json[List[str]]]
    certificate: Optional[str] = None
    cachain: Optional[str] = None


# Properties to receive on creation
class CertificateCreate(CertificateBase):
    pass


# Properties shared by models stored in DB
class CertificateBaseInDB(CertificateBase):
    id: int

    class Config:
        """Pydantic Configuration."""

        orm_mode = True


# Properties to receive via API on update
class CertificateUpdate(CertificateBaseInDB):
    pass
```

The endpoint documentation is OK but when it tries to safe stuff to the DB, with this example of a payload:

```json
{
  "commonname": [
    "string"
  ],
  "certificate": "string",
  "cachain": "string",
}
```

It returns a validation error; before trying to save it to the DB:
```json
{
  "detail": [
    {
      "loc": [
        "body",
        "certificate_in",
        "commonname"
      ],
      "msg": "JSON object must be str, bytes or bytearray",
      "type": "type_error.json"
    }
  ]
}
```

Is this expected behaviour or does the Pydantic ORM mode not support JSON fields in the underlying SQLAlchemy functionality? I wonder what I can do to correctly store the data for this use case. Can I override or implement the behaviour somehow for one field?

**Comment by acidjunk** [comment, marked as answer]
Hurray: when I use

```Python
# Shared properties
class CertificateBase(BaseModel):
    commonname: Optional[List[str]]
    certificate: Optional[str] = None
    cachain: Optional[str] = None
```
It works ok. I was wrongly assuming that I needed a Pydantic JSON type...

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
